opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -10,26 +11,39 @@ use opentelemetry::{global, metrics::MetricsError};
 
 use super::{
     aggregate::{is_under_cardinality_limit, STREAM_OVERFLOW_ATTRIBUTE_SET},
-    Number,
+    AtomicTracker, Number,
 };
 
 /// The storage for sums.
-#[derive(Default)]
 struct ValueMap<T: Number<T>> {
     values: Mutex<HashMap<AttributeSet, T>>,
+    has_no_value_attribute_value: AtomicBool,
+    no_attribute_value: T::AtomicTracker,
```

Comment:
do we expect Histograms to gain from a similar optimization? or the gains won't be that much as we anyway need some locking for histograms anyway?

Reply:
I haven't looked into histograms at all yet.  I just started with counters as they were the part I was most familiarly with.

Reply:
Np. I'll check histograms later.